sources/packages/backend/apps/api/src/route-controllers/report/_tests_/e2e/report.institutions.controller.exportReport.e2e-spec.ts

```diff
@@ -414,4 +437,397 @@ describe("ReportInstitutionsController(e2e)-exportReport", () => {
+            disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+            coeUpdatedAt: new Date(),
+            tuitionRemittanceRequestedAmount: 100,
+          },
+        },
+      );
+      archivedApplication.isArchived = true;
+      await db.application.save(archivedApplication);
+
+      const programYearDefault = notArchivedApplication.programYear;
+      const payload = {
+        reportName: "COE_Requests",
+        params: {
+          offeringIntensity: {
+            "Full Time": true,
+            "Part Time": true,
+          },
+          programYear: programYearDefault.id,
+        },
+      };
+
+      const dryRunSubmissionMock = jest.fn().mockResolvedValue({
+        valid: true,
+        formName: FormNames.ExportFinancialReports,
+        data: { data: payload },
+      });
+      formService.dryRunSubmission = dryRunSubmissionMock;
+      const endpoint = "/institutions/report";
+      const institutionUserToken = await getInstitutionToken(
+        InstitutionTokenTypes.CollegeFUser,
+      );
+
+      // Mock institution user authorization so that the user token will return the fake institution id and mocked roles.
+      await mockInstitutionUserAuthorization(
+        appModule,
+        new InstitutionUserAuthorizations(institution.id, [
+          {
+            locationId: null,
+            userRole: null,
+            userType: InstitutionUserTypes.admin,
+          },
+        ]),
+      );
+
+      // Act/Assert
+      await request(app.getHttpServer())
+        .post(endpoint)
+        .send(payload)
+        .auth(institutionUserToken, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.CREATED)
+        .then((response) => {
+          const fileContent = response.request.res["text"];
+          const parsedResult = parse(fileContent, {
+            header: true,
+          });
+          // Build expected result.
+          const [firstReportDisbursement] =
+            notArchivedApplication.currentAssessment.disbursementSchedules;
+          const firstReportDisbursementData = buildCOERequestsReportData(
+            firstReportDisbursement,
+            "1000.00",
+            getISODateOnlyString(firstReportDisbursement.dateSent),
+          );
+
+          // Expect the disbursement for the application which is not archived.
+          expect(parsedResult.data.length).toBe(1);
+          expect(parsedResult.data).toStrictEqual([
+            firstReportDisbursementData,
+          ]);
+        });
+    },
+  );
+
+  /**
+   * Build COE Requests report data.
+   * @param disbursements disbursements part of the report.
+   * @param expectedDisbursementAmount expected total sum of disbursement amount.
+   * @param expectedDisbursementDate expected disbursement date.
+   * @returns report data.
+   */
+  function buildCOERequestsReportData(
+    disbursement: DisbursementSchedule,
+    expectedDisbursementAmount: string,
+    expectedDisbursementDate: string,
```

Comment:
These two parameters could also be coming from the `disbursement` or was there a reason to prefer them explicitly set as parameters?

Reply:
The reason was to avoid the `buildCOERequestsReportData` doing some logic

e.g 

expectedDisbursementAmount = sum of all awards except BCSG
expectedDisbursementDate = date_sent (if disbursement is sent) else disbursement_date.

Let me know if that is a concern or it still needs to be updated.

Reply:
Not a concern but if the logic will be always the same I would not demand from the method consumer to provide it. The implementation would be clean in the consumer side and the logic to get the values would be centralized in the `buildCOERequestsReportData`.

Reply:
Not a blocker for an E2E test helper.